### ISSUE TITLE
httpfs: S3 multipart upload spawns threads outside our thread pools

### ISSUE THREAD

**Issue by mkornaukhov** [serenedb/serenedb#1058]
## What happens today

The S3 multipart uploader in the vendored `duckdb_httpfs` extension spawns one **raw detached
`std::thread` per uploaded part**, bypassing every thread pool we have (DuckDB's task
scheduler and serenedb's executors):

- `third_party/duckdb_httpfs/src/s3_multi_part_upload.cpp`

  ```cpp
  std::thread upload_thread(S3MultiPartUpload::UploadBuffer, shared_from_this(), write_buffer);
  upload_thread.detach();
  ```

- Concurrency is bounded only by the `s3_uploader_thread_limit` setting
  (`config_params.max_upload_threads`, condvar wait at `s3_multi_part_upload.cpp`),
  default 50 — a global knob with no relation to `threads`, cgroup limits, or current load.
- The only alternative code path is `#define SAME_THREAD_UPLOAD`, which is compile-time and
  enabled solely for EMSCRIPTEN (no threads on WASM). There is no runtime way to keep
  uploads on pool threads.
- Lifetime/errors are hand-rolled around the detached threads: `shared_from_this()` keeps the
  upload state alive after the handle dies, errors are stashed in a `TaskErrorManager` and
  rethrown on the next flush.

This is inherited upstream design, not something we patched in.

## Why it is a problem

1. **Unaccounted threads.** Every writing query can create up to `s3_uploader_thread_limit`
   OS threads that serenedb's scheduler knows nothing about. N concurrent `COPY TO s3://`
   queries → N × limit threads. They compete with the engine's pinned pool for cores and
   distort any `--cpu_threads` sizing we do.
2. **Thread create/destroy per part.** 8 MB parts on a multi-GB COPY means hundreds of
   short-lived threads — pure overhead vs pooled workers.
3. **No integration with cancellation/backpressure.** Detached threads finish on their own
   schedule; query abort only stops *new* parts. Memory for in-flight parts is held by
   threads the engine cannot see (buffer-manager pins held from foreign threads).
4. **Observability.** These threads carry no serenedb thread names/telemetry, complicating
   profiling (they show up as anonymous threads in perf/pstack).

## Proposed fix

Route part uploads through an owned executor instead of `std::thread`:

- Preferred: schedule upload work on DuckDB's `TaskScheduler` (as async tasks with an
  in-flight cap), or on serenedb's IO executor if extension → server dependency is
  acceptable via a hook.
- Keep the existing semantics: bounded in-flight window, order-independent part upload,
  first-error propagation on flush/finalize, `SAME_THREAD_UPLOAD` fallback for WASM.
- Build it as a small reusable "bounded async uploader" utility: the azure write path
  (`duckdb_azure`, `AzureBlobStorageFileHandle::StageWriteBuffer`) currently stages blocks
  serially on the calling thread and is the known bottleneck for single-file `COPY TO az://`;
  when it gets parallelized, it must use this executor rather than repeating the raw-thread
  pattern.
- Keep `s3_uploader_thread_limit` as the in-flight cap, but interpret it as queued tasks,
  not OS threads.

## Acceptance criteria

- No `std::thread` construction in the s3 write path (grep-clean except WASM guard).
- `COPY TO s3://` throughput unchanged or better at `threads=8` (bench:
  `scripts/perf/bench_remote_fs.sh`, write_large / write_many scenarios).
- Thread count during a large COPY stays within engine pool + a bounded IO pool.
- Query cancellation joins/cancels in-flight uploads deterministically.
